libs/wire-api-federation/proto/router.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+package wire.federator;
+
+// These definitions are part of the federation protocol.
+//
+// A component (e.g. 'brig') will send some data of type 'RemoteCall' to
+// the local 'federator' service. That federator will extract the domain from
+// the 'RemoteCall' to discover where to send some data of type 'LocalCall' to.
+// On the other end, the 'LocalCall' will be converted to a 'Response'.
+
+enum Component {
+  Brig = 0;
+}
+
+message HTTPResponse {
+    uint32 responseStatus = 1;
+    bytes responseBody = 2;
+}
+
+message Response {
+  oneof response {
+    HTTPResponse httpResponse = 1;
+    string err = 2;
+  }
+}
+
+// the envelope thing which is sent from brig to local federator
+message RemoteCall {
+  string domain = 1;
+  LocalCall localCall = 2;
+}
+
+// the federator will take out LocalCall from the RemoteCall Envelope, and
+// forward the LocalCall object to a federator on another backend (with the wish
+// that this remote federator forwards this to its own local brig)
+message LocalCall {
```

Comment:
My suggestions for renaming LocalCall and RemoteCall:

RemoteCall --> QualifiedRequest
LocalCall --> UnqualifiedRequest

Reply:
Yeah, sounds great, it will be consistent with rest of the things.

Reply:
renamed as discussed.